### PR TITLE
GEN-664 - refact(ComparisonTableModal): make usage of Mobile/Desktop comparison table components

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Updates `<ComparisonTableModal />` (product page) so it renders the correct version of _Comparison Table_ based on viewport size.

## Justify why they are needed

We got a new [design](https://www.figma.com/file/qUhLjrKl98PAzHov9ilaDH/Hedvig-UI-Kit?type=design&node-id=3380-13894&mode=design&t=snZZXMBD7zxyZULO-4) for a _mobile_ version of the comparison table:
image

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2748** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2748" target="_blank">Graphite</a> 
    * **PR #2749** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2749" target="_blank">Graphite</a> 
      * **PR #2750** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2750" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2750?utm_source=stack-comment).

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2748** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2748" target="_blank">Graphite</a> 
    * **PR #2749** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2749" target="_blank">Graphite</a> 
      * **PR #2750** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2750" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2750?utm_source=stack-comment).

File: apps/store/src/components/ProductPage/PurchaseForm/ComparisonTableModal.tsx [hunk 1]
```diff
@@ -126,6 +102,48 @@ const useGetVeryShortVariantDisplayName = () => {
...
+  }, [tiers, getVeryShortVariantDisplayName])
+
+  const selectedTier = tiers.find((tier) => tier.id === selectedTierId)
+  const selectedTierDisplayName = selectedTier
+    ? getVeryShortVariantDisplayName(selectedTier.variant.typeOfContract)
+    : ''
+
+  if (!selectedTierDisplayName) {
+    console.warn(
+      '[ComparisonTableModal]: could not found tier display name to be used as active table header. Using "" instead.',
```
Comment (robinandeer) [inline apps/store/src/components/ProductPage/PurchaseForm/ComparisonTableModal.tsx:140]: missing some context: include the type of contract in the error message

File: apps/store/src/components/ProductPage/PurchaseForm/ComparisonTableModal.tsx [hunk 2]
```diff
@@ -126,6 +102,48 @@ const useGetVeryShortVariantDisplayName = () => {
...
+
+  const table = useMemo<Table>(() => {
+    const head = [
+      TableMarkers.EmptyHeader,
+      ...tiers.map((tier) => {
+        const headerValue = getVeryShortVariantDisplayName(tier.variant.typeOfContract) ?? ''
+
+        if (!headerValue) {
+          console.warn(
+            '[ComparisonTableModal]: could not found tier display name to be used as table header. Using "" instead.',
```
Comment (robinandeer) [inline apps/store/src/components/ProductPage/PurchaseForm/ComparisonTableModal.tsx:116]: include type of contract in error message

**Review by robinandeer — APPROVED** [review]
Great work!

Gets quite complicated but I wonder if it's because we pass down the unfiltered API data too far into our components 🤔 -- but that's for another PR to address

**Comment by guilhermespopolin** [comment]
@guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2750).

**Comment by guilhermespopolin** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2750) rebased this pull request as part of a merge.

**Comment by guilhermespopolin** [comment]
@guilhermespopolin merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2750).

**Pull request merged**